src/addons/blog/services/blog-sync.ts

```diff
@@ -0,0 +1,254 @@
+    }
+
+    /**
+     * Perform entries syncronization for specified site.
+     *
+     * @param siteId Site id.
+     * @returns Syncronization result.
+     */
+    async syncEntriesForSite(siteId?: string, force: boolean = false): Promise<CoreSyncResult> {
+        const currentSyncPromise = this.getOngoingSync(ADDON_BLOG_SYNC_ID, siteId);
+
+        if (currentSyncPromise) {
+            return currentSyncPromise;
+        }
+
+        const needed = await this.isSyncNeeded(ADDON_BLOG_SYNC_ID, siteId);
+
+        if (force && !needed) {
+            return { updated: false, warnings: [] };
+        }
+
+        const entriesToSync = await this.getPendingToSyncEntries(siteId);
+
+        if (!entriesToSync.length) {
+            return { updated: false, warnings: [] };
+        }
+
+        this.logger.debug('Try to sync ' + ADDON_BLOG_SYNC_ID + ' in site ' + siteId);
+
+        const syncronizations = await Promise.all(entriesToSync.map(async (entry): Promise<CoreSyncResult> => {
+            if (CoreSync.isBlocked(AddonBlogProvider.COMPONENT, entry.created, siteId)) {
+                this.logger.debug('Cannot sync entry ' + entry.created + ' because it is blocked.');
+
+                throw new CoreSyncBlockedError(Translate.instant('core.errorsyncblocked', { $a: this.componentTranslate }));
+            }
+
+            const formattedEntry: AddonBlogAddEntryWSParams & { created: number } = {
+                subject: entry.subject,
+                summary: entry.summary,
+                summaryformat: entry.summaryformat,
+                options: JSON.parse(entry.options),
+                created: entry.created,
+            };
+
+            try {
+                if (entry.id) {
+                    await this.syncUpdatedEntry({ ...entry, id: entry.id, options: JSON.parse(entry.options) });
+
+                    return { updated: true, warnings: [] };
+                }
+
+                const draftId = await this.uploadAttachments({ created: formattedEntry.created, options: formattedEntry.options });
+                const option = formattedEntry.options.find(option => option.name === 'attachmentsid');
+
+                if (draftId) {
+                    option ? option.value = draftId : formattedEntry.options.push({ name: 'attachmentsid', value: draftId });
+                }
+
+                await AddonBlog.addEntry(formattedEntry);
+                await AddonBlogOffline.deleteOfflineEntryRecord(entry.created);
+
+                return { updated: true, warnings: [] };
+            } catch (error) {
+                if (!error || !CoreUtils.isWebServiceError(error)) {
+                    return { updated: false, warnings: [error] };
+                }
+
+                const warnings = [];
+                await AddonBlogOffline.deleteOfflineEntryRecord(entry.created);
+                this.addOfflineDataDeletedWarning(warnings, entry.summary, error);
```

Comment:
```suggestion
                this.addOfflineDataDeletedWarning(warnings, entry.subject, error);
```
I think it's better to use subject, right? Summary is the whole post contents, it can be too long.